src/desdemona/query.clj

```diff
@@ -0,0 +1,25 @@
+(ns desdemona.query
+  (:require [clojure.core.logic :as l]))
+
+(defn ^:private gen-query
+  "Expands a query and events to a core.logic program that executes
+  it."
+  [n-answers query events]
+  `(l/run ~n-answers [results#]
+     (l/fresh [~'x]
+       (l/== [~'x] results#)
+       (l/membero ~'x ~events)
+       ~query)))
```

Comment:
@lvh - I'm ok with this moving forward; but I think it would be helpful to readers to have a comment explaining why the syntax-quoting is necessary and possibly a comment explaining how/why bounding using `n-answers` works.